bootstrap/bootstrap.go

```diff
@@ -52,67 +54,62 @@ type Bootstrap struct {
 }
 
 func New(config *config.Config) (*Bootstrap, error) {
-	logger := zerolog.New(config.LogWriter).With().Timestamp().Logger()
-	logger = logger.Level(config.LogLevel)
-	logger.Info().Msg("starting up the EVM gateway")
+	logger := zerolog.New(config.LogWriter).
+		With().Timestamp().Logger().Level(config.LogLevel)
 
-	// create pebble storage from the provided database root directory
-	store, err := pebble.New(config.DatabaseDir, logger)
+	client, err := setupCrossSporkClient(config, logger)
 	if err != nil {
 		return nil, err
 	}
 
-	// create access client with cross-spork capabilities
-	currentSporkClient, err := grpc.NewClient(config.AccessNodeHost)
-	if err != nil {
-		return nil, fmt.Errorf("failed to create client connection for host: %s, with error: %w", config.AccessNodeHost, err)
-	}
-
-	// if we provided access node previous spork hosts add them to the client
-	pastSporkClients := make([]access.Client, len(config.AccessNodePreviousSporkHosts))
-	for i, host := range config.AccessNodePreviousSporkHosts {
-		grpcClient, err := grpc.NewClient(host)
-		if err != nil {
-			return nil, fmt.Errorf("failed to create client connection for host: %s, with error: %w", host, err)
-		}
-
-		pastSporkClients[i] = grpcClient
-	}
-
-	// initialize cross spork client to the access nodes
-	client, err := requester.NewCrossSporkClient(
-		currentSporkClient,
-		pastSporkClients,
-		logger,
-		config.FlowNetworkID,
-	)
+	storages, err := setupStorage(config, logger)
 	if err != nil {
 		return nil, err
 	}
 
 	return &Bootstrap{
 		publishers: &Publishers{
-			Block:       nil,
-			Transaction: nil,
-			Logs:        nil,
+			Block:       models.NewPublisher(),
+			Transaction: models.NewPublisher(),
+			Logs:        models.NewPublisher(),
 		},
-		storages: &Storages{
-			Storage:      store,
-			Blocks:       pebble.NewBlocks(store, config.FlowNetworkID),
-			Transactions: pebble.NewTransactions(store),
-			Receipts:     pebble.NewReceipts(store),
-			Accounts:     pebble.NewAccounts(store),
-			Traces:       pebble.NewTraces(store),
-		},
-		logger: logger,
-		config: config,
-		client: client,
+		storages: storages,
+		logger:   logger,
+		config:   config,
+		client:   client,
 	}, nil
 }
 
 func (b *Bootstrap) StartEventIngestion(ctx context.Context) error {
 	b.logger.Info().Msg("bootstrap starting event ingestion")
 
+	// get latest cadence block from the network and the database
+	latestCadenceBlock, err := b.client.GetLatestBlock(context.Background(), false)
```

Comment:
Minor, and not really related to this PR, but the `false` argument means that we also take into account non-sealed blocks. Do we maybe want to consider sealed-only blocks?